spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryTests.java

```diff
@@ -33,48 +35,63 @@
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertSame;
 import static org.junit.Assert.assertTrue;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
 /**
  * @author Chengyuan Zhao
+ * @author Balint Pato
  */
 public class SpannerStatementQueryTests {
 
-	@Test
-	public void compoundNameConventionTest() {
+	private static final Object[] EMPTY_PARAMETERS = new Object[0];
+
+	private SpannerOperations spannerOperations;
+
+	private QueryMethod queryMethod;
+
+	private SpannerMappingContext spannerMappingContext;
 
-		QueryMethod queryMethod = mock(QueryMethod.class);
+	private PartTreeSpannerQuery partTreeSpannerQuery;
+
+	@Before
+	public void initMocks() {
+		this.queryMethod = mock(QueryMethod.class);
+		this.spannerOperations = mock(SpannerOperations.class);
+		this.spannerMappingContext = new SpannerMappingContext();
+	}
+
+	private PartTreeSpannerQuery createQuery() {
+		return new PartTreeSpannerQuery(Trade.class,
+				this.queryMethod, this.spannerOperations, this.spannerMappingContext);
```

Comment:
just wondering, why not put this inside initMocks too?

Reply:
I can't :/ because the query creation fails in the constructor - it depends on the `queryMethod.getName()` to be not `null`.

Reply:
LGTM on these refactorings though!